code/modules/overmap/ships/computers/sensors.dm

```diff
@@ -237,10 +237,35 @@
 	return 1
 
 /obj/machinery/shipsensors/update_icon()
-	if(use_power)
-		icon_state = "sensors"
+	icon_state = "sensors_off"
+	if(!use_power)
+		cut_overlays()
+		return
+
+	var/overlay = "sensors-effect"
+
+	var/range_precentage = range / world.view * 100
+
+	if(range_precentage < 20)
```

Comment:
can this be a switch? if not then carry on

Reply:
Cannot be a switch statement.